sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/Beta.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.util;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RetentionPolicy.CLASS)
+@Target({ TYPE, METHOD, PARAMETER, CONSTRUCTOR })
+@Inherited
+/**
+ * Indicates functionality that is in preview and as such is subject to change in non-backwards compatible ways in future releases,
+ * including removal, regardless of any compatibility expectations set by the containing library version.
+ */
+public @interface Beta {
+    String warningText() default "This functionality is in preview and as such is subject to change in non-backwards compatible ways in future releases, including removal, regardless of any compatibility expectations set by the containing library version.";
+
+    /**
+     * @return the version number when the annotated API was first introduced to the library as in Beta
+     */
+    SinceVersion value() default SinceVersion.V4_4_0;
+
+    /**
+     * Azure library version numbers
+     */
+    public enum SinceVersion {
+        /** v4.4.0 */
+        V4_4_0,
+
+        /** v4.5.0 */
+        V4_5_0,
```

Comment:
Ported this straight from Milis's change - i think his intention was to force that only at least Minor versions (not patch versions) bumps are allowed. I kind of liked that model. Do we feel strongly about this?

Reply:
it just sounds weird that we have constants for future unreleased versions here. I would rather to just have only the version we have and we don't add unneeded version constant here.

However not a big deal. no strong feeling. it is your call, whichever you prefer.
Feel free to resolve the comment if you want to keep it they way it is.

Reply:
In the ARM SDK case since we ship this as a separate library, we do need to capture future versions... Not really needed here but it does not hurt either. It's ok either way.

Reply:
Fixed